src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1449,9 +1469,16 @@ void
 mongoc_cursor_set_batch_size (mongoc_cursor_t *cursor, uint32_t batch_size)
 {
    BSON_ASSERT (cursor);
+   bson_iter_t iter;
 
-   _mongoc_cursor_set_opt_int64 (
-      cursor, MONGOC_CURSOR_BATCH_SIZE, (int64_t) batch_size);
+   if (bson_iter_init_find (&iter, &cursor->opts, MONGOC_CURSOR_BATCH_SIZE) &&
+       BSON_ITER_HOLDS_INT32 (&iter)) {
+      _mongoc_cursor_set_opt_int32 (
+         cursor, MONGOC_CURSOR_BATCH_SIZE, (int32_t) batch_size);
```

Comment:
```suggestion
      BSON_ASSERT (bson_in_range_int32_t_unsigned (batch_size));
      bson_iter_overwrite_int32 (&iter, (int32_t) batch_size);
```

Suggest asserting uint32 is in range of int32 before the cast to avoid implementation-defined behavior of unsigned => signed cast.

Suggest removing the `_mongoc_cursor_set_opt_int32` helper, as it is only needed by this caller.